File: src/main/java/com/github/t1/yaml/model/Visitor.java
package com.github.t1.yaml.model;

@SuppressWarnings("unused")
public interface Visitor {
    default void visit(Document document) {}

    default void visitPrefixComment(Comment prefixComment) {}

    default void enterDirectives(Document document) {}

    default void visit(Directive directive) {}

    default void leaveDirectives(Document document) {}

    default void enterBody(Node node) {}

    default void leaveBody(Node node) {}

    default void enterDocumentEnd() {}

    default void visitSuffixCommend(Comment suffixComment) {}

    default void leaveDocumentEnd() {}

    default void leave(Document document) {}


    default void visit(Alias alias) {}


    default void visit(Sequence sequence) {}

    default void enterSequenceItem(Sequence sequence, Sequence.Item item) {}

    default void leaveSequenceItem(Sequence sequence, Sequence.Item item) {}

    default void leave(Sequence sequence) {}


    default void visit(Scalar scalar) {}

    default void enterScalarLine(Scalar node, Scalar.Line line) {}

    default void visit(Scalar.Line line) {}

    default void leaveScalarLine(Scalar node, Scalar.Line line) {}

    default void leave(Scalar scalar) {}


    default void visit(Mapping mapping) {}

    default void enterMappingEntry(Mapping mapping, Mapping.Entry entry) {}

    default void enterMappingKey(Mapping.Entry entry, Node key) {}

    default void leaveMappingKey(Mapping.Entry entry, Node key) {}

    default void enterMappingValue(Mapping.Entry entry, Node key) {}

    default void leaveMappingValue(Mapping.Entry entry, Node key) {}

    default void leaveMappingEntry(Mapping mapping, Mapping.Entry entry) {}

    default void leave(Mapping mapping) {}
}

File: src/main/java/com/github/t1/yaml/model/Visitor.kt
package com.github.t1.yaml.model

interface Visitor {
    fun visit(stream: Stream) {}

    fun leave(stream: Stream) {}


    fun visit(document: Document) {}

    fun visitPrefixComment(prefixComment: Comment) {}

    fun enterDirectives(document: Document) {}

    fun visit(directive: Directive) {}

    fun leaveDirectives(document: Document) {}

    fun enterBody(node: Node) {}

    fun leaveBody(node: Node) {}

    fun enterDocumentEnd() {}

    fun visitSuffixCommend(suffixComment: Comment) {}

    fun leaveDocumentEnd() {}

    fun leave(document: Document) {}


    fun visit(alias: Alias) {}


    fun visit(sequence: Sequence) {}

    fun enterSequenceItem(sequence: Sequence, item: Sequence.Item) {}

    fun leaveSequenceItem(sequence: Sequence, item: Sequence.Item) {}

    fun leave(sequence: Sequence) {}


    fun visit(scalar: Scalar) {}

    fun enterScalarLine(node: Scalar, line: Scalar.Line) {}

    fun visit(line: Scalar.Line) {}

    fun leaveScalarLine(node: Scalar, line: Scalar.Line) {}

    fun leave(scalar: Scalar) {}


    fun visit(mapping: Mapping) {}

    fun enterMappingEntry(mapping: Mapping, entry: Mapping.Entry) {}

    fun enterMappingKey(entry: Mapping.Entry, key: Node) {}

    fun leaveMappingKey(entry: Mapping.Entry, key: Node) {}

    fun enterMappingValue(entry: Mapping.Entry, key: Node) {}

    fun leaveMappingValue(entry: Mapping.Entry, key: Node) {}

    fun leaveMappingEntry(mapping: Mapping, entry: Mapping.Entry) {}

    fun leave(mapping: Mapping) {}
}
